chall05/mtuomine.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+#ruby 2.5.1p57
+
+require 'net/http'
+
+uri = URI("https://en.wikipedia.org/wiki/" + ARGV[0])
+counter = 0
+visited = []
+
+def traverse(visited, uri, counter)
+	html = Net::HTTP.get(uri)
+	if html.empty?
+		puts "no wikipedia page for this word!"
+		exit
+	end
+
+	content = html.split("<p>", 2)
+	linksArr = content[1].scan(/href\s*=\s*"([^"]*)"/)
+
+	# Bubblegum, couldn't combine with regex on tpo
+	res = []
+	linksArr.each { |link|
```

Comment:
if you one-line a each in ruby you write it that way:

```ruby
linksArr.each { |link| puts link }
```

and if you want more lines:

```ruby
linksArr.each do |link|
  puts link
end
```